lib/at_cmd_parser/at_params.c

```diff
@@ -127,7 +127,7 @@ int at_params_short_put(const struct at_param_list *list, size_t index,
 	at_param_clear(param);
 
 	param->type = AT_PARAM_TYPE_NUM_SHORT;
-	param->value.int_val = (uint32_t)(value & USHRT_MAX);
+	param->value.int_val = (int32_t)(value & USHRT_MAX);
```

Comment:
I don't understand the purpose of `value & USHRT_MAX` operation here, the value is already 16 bit, it shouldn't have any effect.

Reply:
Yes, agree

Reply:
Doesn't need the cast either